README.md

```diff
@@ -46,3 +47,31 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+
+# Response
+Got this working now. I could change the unvisited cities to be tracked by an array instead of a set, but I spent about half an hour on it fruitlessly, and would rather move on to other exercises. It would improve the runtime of my implementation significantly though. I have implemented memoization through a cache
+
+## Runtime Analysis
+My implementation follows these steps:
+1. Populate a set with all of the cities
+2. For each starting city, find the shortest tour via steps 3 through 5. Keep the minimum.
+3. For each subset of cities, find the shortest tour via recursion.
+4. To recurse, pass the subset of cities minus one, essentially back to step 3 until there are 2 cities left that have been unvisited. Then, return the distance between them.
+5. Keep the minimum distance of this subset and return it to step 2.
+
+Basically, for a given set of unvisited cities and a start city, it will run $|2^V|$ times to check each unordered permutation of the unvisited cities. Then, whenever it recurses to find subsets, it will pass a copy of the current set of unvisited cities, which is complexity $|V|$. Then, since I want to check for the shortest path starting at each city, that will all run $|V|$ times. This gives an overall complexity of $\Theta\left(2^{|V|}*V^2\right)$
+
+The memory complexity of this algorithm is $|2^V|$, because it will store every version of the unvisited cities set it can come across. Since my implementation sorts the set, there won't be duplicates of the information. The number of possible permutations, disregarding duplicates, is $|2^V|$ because each cities can either be in or out of the set.
```

Comment:
What about the start city?

Reply:
Currently, the way I have the cache set up is to clear after each iteration of my for loop that calls the algorithm for a given set of cities. I suppose that is a time inefficiency because that means there will be subsets I recalculate each time I choose a new starting city from the whole set of cities. This does leave my memory complexity as is, I think.

Reply:
Is that the best way to do the caching? Why are you clearing it?

Reply:
No idea why I was clearing it, all that did was make it take longer. I still need to make sure the cache gets cleared between function calls, but that can happen after the bulk of the algorithm is done running. I've adjusted it now.